infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/KubernetesInternalRuntime.java

```diff
@@ -306,22 +314,38 @@ public void scheduleRuntimeStateChecks() throws InfrastructureException {
    * @throws RuntimeStartInterruptedException when the thread is interrupted while waiting machines
    */
   private void waitMachines(
-      List<CompletableFuture<Void>> machinesFutures,
+      Map<String, CompletableFuture<Void>> machinesFutures,
       List<CompletableFuture<?>> toCancelFutures,
       CompletableFuture<Void> failure)
       throws InfrastructureException {
     try {
       LOG.debug(
           "Waiting to start machines of workspace '{}'",
           getContext().getIdentity().getWorkspaceId());
-      final CompletableFuture<Void> allDone =
-          CompletableFuture.allOf(
-              machinesFutures.toArray(new CompletableFuture[machinesFutures.size()]));
+
+      CompletableFuture<?>[] tracedMachineFutures =
+          machinesFutures
+              .entrySet()
+              .stream()
+              .map(
+                  e ->
+                      e.getValue()
```

Comment:
We can make code more clean with:
```
private CompletableFuture<Void> toFinishMachineSpanFuture(Map.Entry<String, CompletableFuture<Void>> machineNameToFuture) {
...
}

machinesFutures
              .entrySet()
              .stream()
              .map(this::toFinishMachineSpanFuture)
              .toArray(CompletableFuture[]::new);
```

Reply:
Addressed by commit https://github.com/eclipse/che/pull/11924/commits/ce5b4bfef08cc41b999a13708dc16515f63ec042